From 444187b47a87a96ff7ea84841d6927e06a7b09ed Mon Sep 17 00:00:00 2001
From: Simon Waloschek <sonovice@users.noreply.github.com>
Date: Sun, 10 Jan 2021 12:14:07 +0100
Subject: [PATCH] Make `UnknownMetaMessage` robust to faulty MIDIs

For some MIDI files the parameter `type` is passed to the constructor resulting in an TypeError. This workaround should fix this without changing anything else.
---
 mido/midifiles/meta.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/mido/midifiles/meta.py b/mido/midifiles/meta.py
index 54f0765..fbd5f40 100644
--- a/mido/midifiles/meta.py
+++ b/mido/midifiles/meta.py
@@ -540,14 +540,14 @@ def _get_value_names(self):


 class UnknownMetaMessage(MetaMessage):
-    def __init__(self, type_byte, data=None, time=0):
+    def __init__(self, type_byte, data=None, time=0, type='unknown_meta'):
         if data is None:
             data = ()
         else:
             data = tuple(data)

         vars(self).update({
-            'type': 'unknown_meta',
+            'type': type,
             'type_byte': type_byte,
             'data': data,
             'time': time})
